sklearn/semi_supervised/_self_training.py

```diff
@@ -0,0 +1,342 @@
+import warnings
+
+import numpy as np
+
+from ..base import MetaEstimatorMixin, clone, BaseEstimator
+from ..utils.validation import check_is_fitted
+from ..utils.metaestimators import if_delegate_has_method
+from ..utils import safe_mask
+
+__all__ = ["SelfTrainingClassifier"]
+
+# Authors: Oliver Rausch   <rauscho@ethz.ch>
+#          Patrice Becker  <beckerp@ethz.ch>
+# License: BSD 3 clause
+
+
+def _validate_estimator(estimator):
+    """Make sure that an estimator implements the necessary methods."""
+    if not hasattr(estimator, "predict_proba"):
+        msg = "base_estimator ({}) should implement predict_proba!"
+        raise ValueError(msg.format(type(estimator).__name__))
+
+
+class SelfTrainingClassifier(MetaEstimatorMixin, BaseEstimator):
+    """Self-training classifier
+
+    This class allows a given supervised classifier to function as a
+    semi-supervised classifier, allowing it to learn from unlabeled data. It
+    does this by iteratively predicting pseudo-labels for the unlabeled data
+    and adding them to the training set.
+
+    The classifier will continue iterating until either max_iter is reached, or
+    no pseudo-labels were added to the training set in the previous iteration.
+
+    Read more in the :ref:`User Guide <self_training>`.
+
+    Parameters
+    ----------
+    base_estimator : estimator object
+        An estimator object implementing ``fit`` and ``predict_proba``.
+        Invoking the ``fit`` method will fit a clone of the passed estimator,
+        which will be stored in the ``base_estimator_`` attribute.
+
+    criterion : {'threshold', 'k_best'}, optional \
+    (default='threshold')
+        The selection criterion used to select which labels to add to the
+        training set. If 'threshold', pseudo-labels with prediction
+        probabilities above `threshold` are added to the dataset. If 'k_best',
+        the `k_best` pseudo-labels with highest prediction probabilities are
+        added to the dataset.
+        The 'threshold' criterion assumes a well calibrated classifier.
+
+    threshold : float, optional (default=0.75)
+        The decision threshold for use with `criterion`='threshold'.
```

Comment:
```suggestion
        The decision threshold for use with `criterion='threshold'`.
```
To fix the `Inline interpreted text or phrase reference start-string without end-string` warning